src/forcemove.h

```diff
@@ -0,0 +1,98 @@
+     * @param velocities
+     * @param forces
+     */
+    virtual void step(PointVector &velocities, PointVector &forces) = 0; // todo shall we return real time progress?
+    virtual void from_json(const json &j) = 0;
+    virtual void to_json(json &j) const = 0;
+};
+
+void from_json(const json &j, IntegratorBase &i);
+void to_json(json &j, const IntegratorBase &i);
+
+/**
+ * @brief Symmetric Langevin velocity-Verlet method (BAOAB)
+ *
+ * The integrator can conduct normal velocity-Verlet, when gamma = 0.
+ */
+class LangevinVelocityVerlet : public IntegratorBase {
+    float dt;    //!< integration time step (picoseconds)
+    float gamma; //!< friction coefficient (inverse picoseconds)
+
+    //! increment particle's position in an A semi-step
+    inline Point positionIncrement(const Point& velocity);
+    //! increment particle's velocity in a B semi-step
+    inline Point velocityIncrement(const Point& force, const double mass);
+    //! apply fluctuation-dissipation to particle's velocity by solving of Ornstein-Uhlenbeck process in an O-step
+    inline Point velocityFluctuationDissipation(const Point& velocity, const double mass);
+  public:
+    LangevinVelocityVerlet(Space &spc, Energy::Energybase &energy, float dt = 1.0, float gamma = 0.0);
+    LangevinVelocityVerlet(Space &spc, Energy::Energybase &energy, const json &j);
+    void step(PointVector &velocities, PointVector &forces) override;
+    void from_json(const json &j) override;
+    void to_json(json &j) const override;
+};
+
+void from_json(const json &j, IntegratorBase &i);
+void to_json(json &j, const IntegratorBase &i);
+
+/**
+ * @brief Base class for force moves, e.g., molecular dynamics or Langevin dynamics.
+ *
+ * Orchestrate execution of integrators, thermostats, etc. Store vectors for velocities and forces, which are not part
+ * of the particle vector in the space.
+ */
+class ForceMoveBase : public Movebase {
+  protected:
+    unsigned int nsteps; //!< number of integration steps to perform during a single MC move
+    Space &spc;          //!< space with particles and their positions
+    std::shared_ptr<IntegratorBase> integrator;
+    PointVector velocities, forces; //!< velocities and forces; the vector index corresponds to the particle index
+
+    void _to_json(json &j) const override;
+    void _from_json(const json &j) override;
+    void _move(Change &change) override;
+    void generateVelocities(); //!< set the initial conditions for the dynamics: generate velocities and zero forces
+    ForceMoveBase(Space &spc, std::shared_ptr<IntegratorBase> integrator, unsigned int nsteps = 0);
+    virtual ~ForceMoveBase() = default;
+
+  public:
+    double bias(Change &, double, double) override;
+};
+
+/**
+ * @brief Langevin dynamics move using Langevin equation of motion
+ */
+class LangevinDynamics : public ForceMoveBase {
+  public:
+    LangevinDynamics(Space &spc, std::shared_ptr<IntegratorBase> integrator, unsigned int nsteps = 0);
```

Comment:
- Is there a point in giving `nsteps` a default value?
- Remove variable names

Reply:
Nope, done. Only obvious variable names (`spc`) removed.